src/tactic/mk_iff_of_inductive_prop.lean

```diff
@@ -2,34 +2,41 @@
 Copyright (c) 2018 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl
+-/
+import tactic.core
+/-!
+# mk_iff_of_inductive_prop
 
-Generation function for iff rules for inductives, like for `list.chain`:
+This file defines a function that generates `iff` rules for inductive types, like for `list.chain`:
 
-    ∀{α : Type*} (R : α → α → Prop) (a : α) (l : list α),
-      chain R a l ↔ l = [] ∨ ∃{b : α} {l' : list α}, R a b ∧ chain R b l ∧ l = b :: l'
+```lean
+∀{α : Type*} (R : α → α → Prop) (a : α) (l : list α),
+  chain R a l ↔ l = [] ∨ ∃{b : α} {l' : list α}, R a b ∧ chain R b l ∧ l = b :: l'
+```
 
 -/
-import tactic.core
-
 namespace tactic
 
 open tactic expr
 
+/-- Given two expressions `e₀` and `e₁`, return the expression `` `(%%e₀ ↔ %%e₁)``. -/
 meta def mk_iff (e₀ : expr) (e₁ : expr) : expr := `(%%e₀ ↔ %%e₁)
 
-meta def select : ℕ → ℕ → tactic unit
+private meta def select : ℕ → ℕ → tactic unit
```

Comment:
I made this `private` since it seems a bit too specialized to be called `tactic.select`. 

Also, I'd really appreciate it if anyone could help with adding the rest of the missing docstrings!

Reply:
I think it is always better to use a namespace instead of making things `private`. You never know when someone might decide that `tactic.norm_num.not_prime_helper` is actually worth a reference even though you didn't intend it to be broadly usable. :)

Reply:
I'd just namespace the whole file, none of this needs to be in `tactic`, or if it does it can probably move to `tactic.core`.